lib/lib-googlecloud/src/main/java/org/sagebionetworks/googlecloud/SynapseGoogleCloudStorageClientImpl.java

```diff
@@ -70,8 +68,16 @@ public void deleteObject(String bucket, String key) throws StorageException {
 
 	@Override
 	public URL createSignedUrl(String bucket, String key, long expirationInMilliseconds, HttpMethod requestMethod) throws StorageException {
+		Storage.SignUrlOption signatureType;
+		if (requestMethod.equals(HttpMethod.GET)) {
+			// V2 allows us to override content disposition via URL parameter
```

Comment:
Explain in greater detail why this is necessary
- what do you get from v2 vs v4
- what is missing from each?
- why workaround?